### ISSUE TITLE
Integrate the new JsonCpp API instead of the deprecated one

### ISSUE THREAD

**Issue by s3rvac** [avast-tl/retdec#104]
When building RetDec, the compiler reports the following deprecation warnings:
```
src/config/config.cpp:181:21: warning: ‘StyledWriter’ is deprecated: Use StreamWriterBuilder instead [-Wdeprecated-declarations]
src/config/config.cpp:193:15: warning: ‘Reader’ is deprecated: Use CharReader and CharReaderBuilder instead [-Wdeprecated-declarations]
src/fileinfo/file_presentation/json_presentation.cpp:629:15: warning: ‘StyledWriter’ is deprecated: Use StreamWriterBuilder instead [-Wdeprecated-declarations]
```
We should integrate the new JsonCpp API instead of the deprecated one.

**Comment by mbandzi** [comment]
it seems that it won't be that easy as some functionality was [dropped in new API](https://github.com/open-source-parsers/jsoncpp/issues/744). We will have to check if it is easier to adjust our interface or implement missing functionality as mentioned in the link.

**Comment by mbandzi** [comment]
Fixed with [40d83b0b](https://github.com/avast-tl/retdec/commit/40d83b0bf2f5aa91d32f9f7e3fb7a30ec5d0d9e8) and [0e8f16d16](https://github.com/avast-tl/retdec/commit/0e8f16d167765d3f1a0c6d0e4e490f30283f5a14) commits.
